### ISSUE TITLE
Handle infinite loop involving Self Service + The Daily Courant

### ISSUE THREAD

**Issue by micahstairs** [micahstairs/bga-innovation#305]
**NOTE:** Before you begin working on this, please **assign this issue to yourself**. If you do not do this, then there's the possibility that 2 people will end up doing the same task, which is not a good use of time.

If Self Service and The Daily Courant are the only top cards on your board, and you activate one of them, you are forced into an infinite loop with no way out.

<img width="1423" alt="Screen Shot 2022-03-02 at 6 59 03 PM" src="https://user-images.githubusercontent.com/7231485/156472142-d65b26e9-ed4c-4080-afc2-3d4c4077d7fe.png">

It seems that we need to provide a mechanism to abort this loop. But we also can't abort automatically because the player may wish to look at all of the top cards in the decks.

We should only provide a way to abort if these are the only two cards on their board.

I couldn't find any threads talking about this particular situation.

**Comment by micahstairs** [comment]
Marking this as blocked until we hear back from Chris. My proposal was to adjust the wording on The Daily Courant so that executing another card is optional.

**Comment by micahstairs** [comment]
My last email to Chris:

```
So the Self Service + The Daily Courant infinite is actually a bit differently than I explained in a previous email. If a player only has those two cards as their top cards and they execute Self Service for example, their only option to execute is The Daily Courant. And when they execute The Daily Courant they get to choose Self Service to execute. But this time, since Self Service is executing as if it were on The Daily Courant, the only card that can be chosen is Self Service (again). This leads to a The Daily Courant > Self Service (as if on The Daily Courant) > Self Service infinite loop.

Similarly, the other infinite loop I described is a bit different (Gujin Tushu Jinsheng > The Daily Courant > Self Service (as if on The Daily Courant) > Self Service).

Upon further reflection, I think it's okay if we allow the possibility of these infinite loops. On BGA, players have the option to abandon the game (it might make players complain about losing ELO). However, I expect the odds of these infinite loops from actually happening in practice to be extremely low. If they have any other other top cards on their board, then the loop can be broken.

I'd still be interested in hearing Carl's thoughts about these infinite loops, but I think it's fine if we don't give these infinite loops special treatment in the implementation.
```

**Comment by micahstairs** [comment]
Chris' response:

```
So I'm torn on this. On one hand, infinite loops (especially on a platform like BGA that has ELO concerns with abandoning games) is messy. But on the other, this is *so* unlikely to happen. I suppose it's slightly more likely in a Fission game.

Carl's actually OK with doing the "you may" modification for digital, and possibly revising the card next time we do a print run. I'm fine either way, really. I think the impact of the change will be extremely small. 
```

**Comment by micahstairs** [comment]
My response:

> If Carl is onboard with the minor revision of "The Daily Courant", then I'm going to move forward with that as our solution!
> 
> **Before:** Draw a card of any value, then place it on top of the draw pile of its age. Execute the effects of one of your other top cards as if they were on this card. Do not share them.
> **After:** Draw a card of any value, then place it on top of the draw pile of its age. You may execute the effects of one of your other top cards as if they were on this card. Do not share them.
